tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend

```diff
@@ -117,15 +128,43 @@ class DropUtils {
 			return (dropTarget.eContainer as SpecificationStepImplementationImpl).getContexts().
 				head as ComponentTestStepContext
 		}
-
+		if (dropTarget instanceof MacroTestStepContextImpl || (dropTarget instanceof TestStepImpl &&
+			dropTarget.eContainer instanceof MacroTestStepContextImpl)) {
+			val macroTestStepContext = EcoreUtil2.getContainerOfType(dropTarget, MacroTestStepContext)
+			val EList<TestStepContext> contexts = EcoreUtil2.getContainerOfType(dropTarget, StepContainer).contexts
+
+			var contextIndex = contexts.indexOf(macroTestStepContext)
+			while (contextIndex < contexts.size()) {
+				if (contexts.get(contextIndex) instanceof ComponentTestStepContext) {
+					return contexts.get(contextIndex) as ComponentTestStepContext
+				}
+				contextIndex++
+			}
+			return null
+		}
 		return EcoreUtil2.getContainerOfType(dropTarget, ComponentTestStepContext)
 	}
 
+	protected def StepContainer getLastStepContext(TclModel tclModel) {
+		if (tclModel.test != null) {
```

Comment:
if (tclModelTest.test !== null) {
  return tclMode.test.steps.last // <- is null if collection is empty
}